f8a_worker/storages/s3.py

```diff
@@ -212,6 +212,13 @@ def retrieve_latest_version_id(self, object_key):
 
         return self._s3.Object(self.bucket_name, object_key).version_id
 
+    def list_available_task_results(self, arguments):
+        """Get names of all task results stored on S3."""
+        object_key = self._construct_base_file_name(arguments)
+        bucket = self._s3.Bucket(self.bucket_name)
+        objects = bucket.objects.filter(Prefix=object_key)
+        return list(x.key for x in objects if x.key.endswith('.json'))
```

Comment:
@miteshvp This will probably list task results with the whole object key ("path"). If you don't want to do it that way, feel free to adjust to return only task name.